### PR TITLE
Bug/e UI 8965 post registrators

### PR DESCRIPTION
**Pull request by connorpgpmcelroy**
### Jira link (if applicable)
https://tools.hmcts.net/jira/browse/EUI-8965

### Change description ###
Add the correct details onto the MO organisation screens

### CODE REVIEW

File: api/registerOrganisation/index.ts [hunk 1]
```diff
@@ -38,6 +38,22 @@ export function mapRequestObject(requestBody: RegistrationData): RegistrationReq
...
+  if (requestBody.regulators && requestBody.regulators.length > 0) {
+    for (let i = 0; i < requestBody.regulators.length; i++) {
+      request.orgAttributes.push({
+        key: `regulators-${i}`,
+        value: JSON.stringify(requestBody.regulators[i])
+      });
+    }
+  }
+  if (requestBody.individualRegulators && requestBody.individualRegulators.length > 0) {
+    for (let i = 0; i < requestBody.individualRegulators.length; i++) {
```
Comment (udaydenduluri33) [inline api/registerOrganisation/index.ts:50]: Please replace it with map function

File: src/organisation/utils/index.ts [hunk 2]
```diff
@@ -27,14 +27,19 @@ const utils = {
...
     }
     return null;
   },
   getRegulators: (organisationDetails: Partial<OrganisationDetails>): Regulator[] => {
-    if (containsItems(organisationDetails, 'regulators')) {
-      return organisationDetails.regulators;
+    if (containsItems(organisationDetails, 'orgAttributes') && organisationDetails.orgAttributes.find((orgAttribute) => orgAttribute.key.includes('regulator'))) {
+      const regulatorAttributes = organisationDetails.orgAttributes.filter((orgAttribute) => orgAttribute.key.includes('regulator'));
+      const regulators = [];
+      for (let i = 0; i < regulatorAttributes.length; i++) {
```
Comment (udaydenduluri33) [inline src/organisation/utils/index.ts:39]: Again Please replace it with map function

File: api/organisation/index.ts [hunk 3]
```diff
@@ -21,7 +21,24 @@ export async function handleOrganisationRoute(req: Request, res: Response) {
...
+// delete next two functions after register org feature not necessary
+export async function handleOrganisationV1Route(req: Request, res: Response) {
+  try {
+    const response = await req.http.get(
+      `${getConfigValue(SERVICES_RD_PROFESSIONAL_API_PATH)}/refdata/external/v1/organisations`
+    );
+    res.send(response.data);
+  } catch (error) {
+    const errReport = {
+      apiError: error.data.message,
```
Comment (udaydenduluri33) [inline api/organisation/index.ts:36]: I think its better to use next [global error handler] instead of handling the error here.

Comment (connorpgpmcelroy) [reply api/organisation/index.ts:36]: not actually my code, just copied and pasted, but I will try to resolve this anyway

File: api/registerOrganisation/index.ts [hunk 4]
```diff
@@ -38,6 +38,26 @@ export function mapRequestObject(requestBody: RegistrationData): RegistrationReq
...
+    requestBody.regulators.map((regulator) => {
+      request.orgAttributes.push({
+        key: `regulators-${regIndex}`,
+        value: JSON.stringify(regulator)
+      });
+      regIndex++;
+    });
+  }
+  if (requestBody.individualRegulators && requestBody.individualRegulators.length > 0) {
+    let iRegIndex = 0;
```
Comment (udaydenduluri33) [inline api/registerOrganisation/index.ts:52]: map function provides a function by default. You dnt need to add an index for it
`const numbers = [1, 2, 3, 4];
const filteredNumbers = numbers.map((num, index) => {
  if (index < 3) {
    return num;
  }
});`

**Review by udaydenduluri33 — DISMISSED** [review]

**Review by DavidJayakumar — APPROVED** [review]
LGTM

**Pull request merged**